### DISCUSSION TITLE
Codecov when there is switch case with random

### DISCUSSION THREAD

**Discussion by 00thirdeye00** [cieslarmichal/faker-cxx#798] [General]
[company test](https://app.codecov.io/github/cieslarmichal/faker-cxx/blob/main/tests%2Fmodules%2Fcompany%2Fcompany_test.cpp) for example has multiple options to generate company name. this happens in random fashion which won't cover code. my suggestion is to change such functions to have a default parameter. default parameter being no parameter. if no parameter is given run it in random fashion (like the current implementation) otherwise generate name as per the parameter passed.

**Comment by cieslarmichal** [comment]
we could introduce  `CompanyNameFormat` enum param that user could choose the company name format for example `CompanyNameFormat::LastNameSuffix` and `CompanyNameFormat::FirstNameLastNameSuffix` etc

**Reply by 00thirdeye00** [reply]
Okej I will look into that today

**Reply by 00thirdeye00** [reply]
issues fixed: #834

**Reply by cieslarmichal** [reply]
Nice
